File: Algorithm/Programers/programers_42840/programers_42840.py
# https://programmers.co.kr/learn/courses/30/lessons/42840

def solution(answers):
    answer = [0, 0, 0, 0]
    result = []
    student = [[1, 2, 3, 4, 5],
               [2, 1, 2, 3, 2, 4, 2, 5],
               [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]]
    maxscore = 0

    for s in range(3):
        score = 0
        j = 0
        for i in range(len(answers)):
            if(s == 0):
                j = i % 5
            elif(s == 1):
                j = i % 8
            else:
                j = i % 10
            if answers[i] == student[s][j]:
                score = score + 1
        if score >= maxscore:
            maxscore = score
            answer[s] = maxscore

    for i in range(3):
        print(i, "=", answer[i])
    if answer[i] == maxscore:
        result.append(i+1)

    return result

File: Algorithm/Programers/p42840/programers_42840.js
// https://programmers.co.kr/learn/courses/30/lessons/42840

function solution(answers) {
    var answer = [];
    var user = 
        [[1,2,3,4,5], 
        [2,1,2,3,2,4,2,5],
        [3,3,1,1,2,2,4,4,5,5]];
    var point = [0,0,0];

    for(var i=0; i<answers.length; i++) {
        if(user[0][i%5] == answers[i])  point[0]++; 
        if(user[1][i%8] == answers[i])  point[1]++; 
        if(user[2][i%10] == answers[i]) point[2]++; 
    }

    var max = Math.max(... point);

    for(var i=0; i<3; i++) {
        if(point[i] == max) answer.push(i+1);
    }

    return answer;
}
